pkg/build/api/validation/validation.go

```diff
@@ -54,6 +55,29 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) field.Err
 	return allErrs
 }
 
+// ValidateBuildDetailsUpdate validates build details update
+func ValidateBuildDetailsUpdate(build *buildapi.Build, older *buildapi.Build) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if older.Spec.Revision != nil {
+		// If there was already a revision, then return an error
+		allErrs = append(allErrs, field.Duplicate(field.NewPath("status", "revision"), older.Spec.Revision))
+	}
+	if build.Spec.Revision == nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("status", "revision"), nil, "cannot set an empty revision in build status"))
+	}
+	return allErrs
+}
+
+// ValidateBuildStatusUpdate validates a build status update
+func ValidateBuildStatusUpdate(build *buildapi.Build, older *buildapi.Build) field.ErrorList {
+	allErrs := field.ErrorList{}
```

Comment:
the base store does that for you outside of this call
https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/rest/update.go#L93